utils/build-script-impl

```diff
@@ -3181,6 +3181,7 @@ for host in "${ALL_HOSTS[@]}"; do
                 )
 
                 [[ -z "${DISTCC}" ]] || EXTRA_DISTCC_OPTIONS=("DISTCC_HOSTS=localhost,lzo,cpp")
+                export CTEST_OUTPUT_ON_FAILURE=1
```

Comment:
@spevans This is going to export it for the rest of build-script-impl's invocation. I think it may cause weird confusion for people unless it is put in global scope at the top of build-script-impl with a nice comment. That or put it on the actual test command for Foundation below.

Can you make that change?

Reply:
@gottesmm I did originally try and enable it in `swift-corelibs-foundation`'s cmake config but could not get it to work, does the following look ok?

```diff --git a/utils/build-script-impl b/utils/build-script-impl
index e7bf8f734b..0d619392fc 100755
--- a/utils/build-script-impl
+++ b/utils/build-script-impl
@@ -2853,6 +2853,10 @@ for host in "${ALL_HOSTS[@]}"; do
     # Calculate test targets
     calculate_targets_for_host $host
 
+    # If the tests for a product fail, make ctest output the full test output
+    # instead of just printing that some of the tests failed.
+    export CTEST_OUTPUT_ON_FAILURE=1
+
     # Run the tests for each product
     for product in "${PRODUCTS[@]}"; do
         # Check if we should perform this action.
@@ -3181,7 +3185,6 @@ for host in "${ALL_HOSTS[@]}"; do
                 )
 
                 [[ -z "${DISTCC}" ]] || EXTRA_DISTCC_OPTIONS=("DISTCC_HOSTS=localhost,lzo,cpp")
-                export CTEST_OUTPUT_ON_FAILURE=1
                 with_pushd "$(build_directory ${host} foundation)" \
                     call env "${EXTRA_DISTCC_OPTIONS[@]}" "${CMAKE}" "${cmake_options[@]}" "${EXTRA_CMAKE_OPTIONS[@]}" "${FOUNDATION_SOURCE_DIR}"
 
```